Implement @Condition enricher

LebombJames opened the pull request:
- Refactor custom enrichers to bring them inline with V12/V13 changes
- Also implement support for event listeners other than click

`@Condition` mandated the implementation of a cache for condition items. This should probably be used to add conditions to actors, instead of always fetching from the compendiums

LebombJames committed:
Implementation of Condition enricher

LebombJames committed:
Whoops

LebombJames committed:
Revert changes because I cba

ian612 commented on src/module/system/enrichers/base.js:
```diff
@@ -189,45 +180,34 @@ export default class BaseEnricher {
-    }
+export function getDatasetfromEvent(event) {
+    const anchorEl = event.currentTarget.children[0];
+    return anchorEl?.dataset || null;
```
This doesn't seem to be getting any `duration` or `unit` fields from the dataset, so they aren't being set

LebombJames committed:
- Fix @Condition, implement turn events
- Move document icons to respective data models, and register them programmatically
- Move compendium browser indexing to data models, register indexable fields programmatically
- Migrate all references to SFRPG global to CONFIG.SFRPG, rename SFRPG to _SFRPG to discourage use
- A bunch of typing

LebombJames committed:
Merge branch 'development' into enricher-events

LebombJames committed:
Correctly mark new weapon fields

LebombJames committed:
Correctly set condition status on actors when dragging from links

LebombJames committed:
Centralise @condition damage type parsing

LebombJames committed:
More types changes

LebombJames committed:
V14 fixes

LebombJames committed:
Change message `type` to `style`

danimrath commented on src/module/token/token-ruler.js:
```diff
@@ -119,7 +119,7 @@ export default class SFRPGTokenRuler extends foundry.canvas.placeables.tokens.To
                 }
             }
-        }, { performDeletions: true });
+        }, { applyOperators: true });
```
What does this change do?
